openerp/openupgrade/openupgrade.py

```diff
@@ -135,6 +137,31 @@ def load_data(cr, module_name, filename, idref=None, mode='init'):
 table_exists = openupgrade_tools.table_exists
 
 
+def copy_columns(cr, column_spec):
+    """
+    Copy table columns. Typically called in the pre script.
+
+    :param column_spec: a hash with table keys, with lists of tuples as \
+    values. Tuples consist of (old_name, new_name, type).  Use None for \
```

Comment:
When the copy is done, the type is still the same as far as I know (it is still date), because it is executed on pre-script. Anyways, we can have it as an optional parameter, that if not filled, the type is infered, and if set, the type is overwritten.